### PR TITLE
Std migration for base

### PR DESCRIPTION
**Pull request by milchakov**

### CODE REVIEW

**Comment by milchakov** [comment]
DUDE JTTP

**Comment by milchakov** [comment]
DUDE JTTP

**Comment by milchakov** [comment]
DUDE JTTP

**Comment by milchakov** [comment]
@mpimenov @ygorshenin PTAL

**Comment by milchakov** [comment]
@mpimenov @ygorshenin PTAL

File: base/base_tests/stl_helpers_test.cpp [hunk 1]
```diff
@@ -61,9 +61,9 @@ template <template <typename...> class Cont>
 void TestEqualsBy()
 {
   {
-    using Value = pair<int, int>;
+    using Value = std::pair<int, int>;
```
Comment (ygorshenin) [inline base/base_tests/stl_helpers_test.cpp]: Why not to just put "using namespace std" at the top of source files? The number of changes will be significantly smaller.

File: base/base_tests/regexp_test.cpp [hunk 2]
```diff
@@ -3,73 +3,73 @@
...
+#include <regex>
 
 UNIT_TEST(RegExp_Or)
 {
-  regex exp("\\.mwm\\.(downloading2?$|resume2?$)");
+  std::regex exp("\\.mwm\\.(downloading2?$|resume2?$)");
 
-  TEST(regex_search("Aruba.mwm.downloading", exp), ());
+  TEST(std::regex_search("Aruba.mwm.downloading", exp), ());
   TEST(!regex_search("Aruba.mwm.downloading1", exp), ());
```
Comment (mpimenov) [inline base/base_tests/regexp_test.cpp]: ?

**Pull request merged**